### PR TITLE
api: add device info into NodeMetric CRD

### PR DESCRIPTION
**Pull request by jasonliu747**
Signed-off-by: Jason Liu <jasonliu747@gmail.com>

### Ⅱ. Does this pull request fix one issue?
Prerequisite for #370

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

File: apis/slo/v1alpha1/common.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	schedulingv1alpha1 "github.com/koordinator-sh/koordinator/apis/scheduling/v1alpha1"
+)
+
+type ResourceMap struct {
+	corev1.ResourceList `json:"resources,omitempty"`
+	Devices             []schedulingv1alpha1.DeviceInfo `json:"devices,omitempty"`
```
Comment (zwzhang0107) [inline apis/slo/v1alpha1/common.go:27]: for usage report? `total capacity`comments of DeviceInfo.Resource at "https://github.com/koordinator-sh/koordinator/blob/91cacc405bb286a20d0af4f921ca779f867faded/apis/scheduling/v1alpha1/device_types.go#L45" needs modify

File: apis/slo/v1alpha1/common.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	schedulingv1alpha1 "github.com/koordinator-sh/koordinator/apis/scheduling/v1alpha1"
+)
+
+type ResourceMap struct {
+	corev1.ResourceList `json:"resources,omitempty"`
```
Comment (zwzhang0107) [inline apis/slo/v1alpha1/common.go:26]: make corev1.ResourceList as inline for json?

File: apis/slo/v1alpha1/common.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
+/*
```
Comment (zwzhang0107) [inline apis/slo/v1alpha1/common.go:1]: common.go->resources.go

File: apis/extension/resource.go [hunk 4]
```diff
@@ -28,6 +28,12 @@ const (
 	BatchCPU    corev1.ResourceName = DomainPrefix + "batch-cpu"
 	BatchMemory corev1.ResourceName = DomainPrefix + "batch-memory"
 
+	GPUCore     corev1.ResourceName = DomainPrefix + "gpu-core"
+	GPUMem      corev1.ResourceName = DomainPrefix + "gpu-memory"
+	GPUMemRatio corev1.ResourceName = DomainPrefix + "gpu-memory-ratio"
```
Comment (jasonliu747) [inline apis/extension/resource.go:33]: `GPUMem` or `GPUMemory`
`GPUMemRatio` or `GPUMemoryRatio`
Please let me know what you guys think.

Comment (eahydra) [reply apis/extension/resource.go:33]: `GPUMemory` And `GPUMemoryRatio` +1

Comment (jasonliu747) [reply apis/extension/resource.go]: image

**Pull request merged**